### PR TITLE
Bump HdrHistogram to 2.2.2 and move the dependency version to version.properties

### PR DESCRIPTION
**Pull request by LantaoJin**
### Description
1. Bump org.hdrhistogram:HdrHistogram from 2.2.1 to 2.2.2 in /server.
Sourced from [org.hdrhistogram:HdrHistogram's releases](https://github.com/HdrHistogram/HdrHistogram/releases).

>##  HdrHistogram-2.2.2
>[Revert moditect plugin use and use Automatic-Module-Name instead](https://github.com/HdrHistogram/HdrHistogram/commit/c655bfa1654158157605cf30987c4e5e5d8c7b2b)
>Fixes [HdrHistogram/HdrHistogram#206](https://redirect.github.com/HdrHistogram/HdrHistogram/issues/206)

2. Add the dependency versions of `t-digest` and `hdrhistogram` to version.properties and fix a typo in comment.
https://github.com/opensearch-project/OpenSearch/pull/3634 upgraded t-digest to 3.3 in main only, but keep using 3.2 in 2.x branch. SQL plugin which depends on these percentile aggregation dependencies has to set different versions for each branch. Move the versions to `version.properties` file could reduce the dependency maintenance efforts in plugins.

### Related Issues
Followup https://github.com/opensearch-project/OpenSearch/pull/3634
Duplicates https://github.com/opensearch-project/OpenSearch/pull/13937

### Check List
 - ~~[ ] New functionality includes testing.~~
  - ~~[ ] All tests pass~~
- ~~[ ] New functionality has been documented.~~
  - ~~[ ] New functionality has javadoc added~~
- ~~[ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md).~~
- [x] Commits are signed per the DCO using --signoff
- ~~[ ] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)~~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by LantaoJin** [comment]
Changelog could be skip

File: server/build.gradle [hunk 1]
```diff
@@ -96,9 +96,9 @@ dependencies {
   api "joda-time:joda-time:${versions.joda}"
 
   // percentiles aggregation
-  api 'com.tdunning:t-digest:3.3'
-  // precentil ranks aggregation
-  api 'org.hdrhistogram:HdrHistogram:2.2.1'
+  api "com.tdunning:t-digest:${versions.tdigest}"
+  // percentile ranks aggregation
+  api "org.hdrhistogram:HdrHistogram:${versions.hdrhistogram}"
```
Comment (gaobinlong) [inline server/build.gradle:101]: This change conflicts with the [PR](hdrhistogram) which bumps `hdrhistogram` to 2.2.2, maybe we can bump the version in this PR and close that older PR.

Comment (LantaoJin) [reply server/build.gradle:101]: Sure, let me refactor this one

**Comment by reta** [comment]
> ❌ Gradle check result for [c83e229](https://github.com/opensearch-project/OpenSearch/commit/c83e2290eff291218608301a2b7e2f8b87d1aff8): [FAILURE](https://build.ci.opensearch.org/job/gradle-check/39949/)

```
java.lang.Exception: Suite timeout exceeded (>= 1200000 msec).
	at __randomizedtesting.SeedInfo.seed([E01B74F9814450C]:0)
```

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
@LantaoJin could you please backport to `2.x` manually? thank you

**Comment by LantaoJin** [comment]
@reta sure, I will backport this to 2.x.

**Pull request merged**